plugins/Marketplace/resources/qml/PackageCardHeader.qml

```diff
@@ -32,8 +32,24 @@ Item
         }
         width: UM.Theme.getSize("card_icon").width
         height: width
-
-        source: packageData.iconUrl != "" ? packageData.iconUrl : "../images/placeholder.svg"
+        sourceSize.height: height
+        sourceSize.width: width
+        source:
+        {
+            if(packageData.iconUrl != "")
+            {
+                return packageData.iconUrl
+            }
+            if(packageData.packageType == "plugin")
+            {
+                return "../images/Plugin.svg"
+            }
+            if(packageData.packageType == "material")
+            {
+                return "../images/Spool.svg"
+            }
+            return "../images/placeholder.svg"
```

Comment:
I think this would be a great candidate for a switch statement
```suggestion
            switch (packageData.packageType)
            {
                case "plugin":
                    return "../images/Plugin.svg";
                case "material":
                    return "../images/Spool.svg";
                default:
                    return "../images/placeholder.svg";
            }
```

Reply:
For compactness, in personal project I also write something like
```suggestion
            switch (packageData.packageType)
            {
                case "plugin": return "../images/Plugin.svg";
                case "material": return "../images/Spool.svg";
                default: return "../images/placeholder.svg";
            }
```
I like this as the low complexity is better reflected in the number of lines used. (wasting a large number of lines on something that isn't very difficult to understand is a bit annoying to read imo)